portal/src/components/FilterCategories/FilterCategories.component.js

```diff
@@ -36,76 +37,47 @@ export default {
         this.$forceUpdate()
       }
     },
-    onToggleShowAll(category) {
-      category.showAll = !category.showAll
-      this.$forceUpdate()
-    },
-    /**
-     * Set filter for v-model
-     * @returns {*} - filters
-     */
-    setFilter() {
-      const { categories, publisherdate } = this
-      if (categories) {
-        let filters = []
-
-        const publisherdate_item = this.value.filters.find(
-          item => item.external_id === publisherdate
-        )
-
-        if (publisherdate_item) {
-          filters.push({
-            external_id: publisherdate,
-            items: [...publisherdate_item.items]
-          })
-          this.data = {
-            start_date: publisherdate_item.items[0],
-            end_date: publisherdate_item.items[1]
-          }
-        }
-
-        filters = Object.assign({}, this.value, {
-          filters
-        })
-
-        this.$router.push(this.generateSearchMaterialsQuery(filters))
+    onChange(e) {
+      const { categoryId, itemId } = e.target.dataset
+      const filter = { external_id: categoryId, items: [itemId] }
+      let filters = this.selectedFilters
 
-        this.$emit('input', filters)
-
-        return filters
+      if (e.target.checked) {
+        filters.push(filter)
+      } else {
+        filters.splice(filters.indexOf(filter), 1)
       }
+      this.executeSearch(filters)
     },
-    onChange(event, parent) {
-      if (!_.isNil(parent)) {
-        parent.selected = !parent.selected
+    onDateChange(dates) {
+      const { start_date, end_date } = dates
+      const filter = {
+        external_id: this.publisherDate,
+        items: [start_date, end_date]
       }
-      this.$store.commit('SET_FILTER_SELECTED', event.target.dataset.categoryId)
-      this.executeSearch()
-    },
-    onDateChange() {
-      this.executeSearch()
+      let filters = this.selectedFilters.filter(
+        el => el.external_id !== this.publisherDate
+      )
+      filters.push(filter)
+      this.executeSearch(filters)
     },
-    executeSearch() {
-      let searchText = this.$store.getters.materials.search_text
-      let ordering = this.$store.getters.materials.ordering
+    executeSearch(filters = []) {
+      const { ordering, search_text } = this.materials
       let searchRequest = {
-        search_text: searchText,
+        search_text: search_text,
```

Comment:
hm, niet searchText is meer javascript naming. En anders zijn die dubbele punten niet nodig, want de object keys zijn hetzelfde als de variabele naam

Reply:
Django backend geeft default snake case terug, dus zou daar wel een uitzondering voor willen maken, anders moet heel de backend om en ik heb het eerlijk gezegd ook nooit hoeven doen, dus weet niet hoe makkelijk dit gaat. Voor JS variabelen is camel case prima wat mij betreft.

Reply:
Ja, dat bedoel ik 😉